### DISCUSSION TITLE
Pysolarmanv5 Integration

### DISCUSSION THREAD

**Discussion by Xhanti** [kellerza/sunsynk#143] [Ideas]
Hi,

Thank you for the excellent library and homeassitant add on, it's been a treat reading the code. I've been experimenting with extending this library to work with a solarman dongle (so no cables needed, talk over wifi) and had some success using [this](https://github.com/jmccrohan/pysolarmanv5). As far as  i can tell, all one needs to implement is one of [these](https://github.com/kellerza/sunsynk/blob/main/sunsynk/sunsynk.py) like [here](https://github.com/kellerza/sunsynk/blob/main/sunsynk/usunsynk.py). Am I on the right track? And is this a contribution this project is interested in accepting.

Regards

**Comment by kellerza** [comment]
Hi! This looks very interesting. It seems like they read the same underlying registers, so it should be as easy as adding a new subclass.

Currently only the multi addon is being updated, so you need some driver support [here](https://github.com/kellerza/sunsynk/blob/main/hass-addon-sunsynk-multi/run.py#L108). And then `tcp://` probably makes sense as the url scheme?

**Reply by Xhanti** [reply]
Hi, thank you for the quick response. 
It doesn't seem too bad. 
I'll take a stab at it this weakened. 
I just need to figure out how to test my changes locally .

**Reply by kellerza** [reply]
On Windows you can run the pymosdbus driver code locally (it loads settings from `.local.yaml`). Not umodbus, I expect pysolarman will be the same since it relies on umodbus, but you can try. Probably using wsl2 you could run it locally

You can also use `scripts\copy2local.cmd` to copy to you HASS OS - I do this in standard windows + msys2 for the linux tools - there you can compile and run the addon

**Reply by Xhanti** [reply]
Thanks. 
I'll play around and see where I land. 
I have linux and windows boxes here, so I should come right.
